l10n_it_edi_extension/models/account_move.py

```diff
@@ -0,0 +1,874 @@
+                self.l10n_it_edi_amount_untaxed += get_float(
+                    element_line, ".//PrezzoTotale"
+                )
+
+        if elements_summary := body_tree.xpath(".//DatiBeniServizi/DatiRiepilogo"):
+            for element_summary in elements_summary:
+                self.env["l10n_it_edi.summary_data"].create(
+                    {
+                        "tax_rate": get_float(element_summary, ".//AliquotaIVA"),
+                        "non_taxable_nature": get_text(element_summary, ".//Natura"),
+                        "incidental_charges": get_float(
+                            element_summary, ".//SpeseAccessorie"
+                        ),
+                        "rounding": get_float(element_summary, ".//Arrotondamento"),
+                        "amount_untaxed": get_float(
+                            element_summary, ".//ImponibileImporto"
+                        ),
+                        "amount_tax": get_float(element_summary, ".//Imposta"),
+                        "payability": get_text(element_summary, ".//EsigibilitaIVA"),
+                        "law_reference": get_text(
+                            element_summary, ".//RiferimentoNormativo"
+                        ),
+                        "invoice_id": self.id,
+                    }
+                )
+                self.l10n_it_edi_amount_tax += get_float(element_summary, ".//Imposta")
+
+        extra_info["l10n_it_edi_ext_body_tree"] = body_tree
+        return extra_info, message_to_log
+
+    def _l10n_it_edi_create_partner(
+        self, xml_tree, partner_section_xpath, vat, codice_fiscale
+    ):
+        country_id = False
+        is_company = bool(get_text(xml_tree, partner_section_xpath + "//Denominazione"))
+        eori_code = get_text(xml_tree, partner_section_xpath + "//CodEORI")
+
+        if country_code := get_text(xml_tree, partner_section_xpath + "//IdPaese"):
+            countries = self.env["res.country"].search([("code", "=", country_code)])
+            if countries:
+                country_id = fields.first(countries).id
+            else:
+                raise UserError(
+                    self.env._("Country Code %s not found in system.") % country_code
+                )
+
+        vals = {
+            "vat": country_code + vat,
+            "l10n_it_codice_fiscale": codice_fiscale,
+            "is_company": is_company,
+            "l10n_edi_it_eori_code": eori_code,
+            "country_id": country_id,
+        }
+
+        if value := get_text(xml_tree, partner_section_xpath + "//Denominazione"):
+            vals["name"] = value
+        else:
+            vals["name"] = " ".join(
+                filter(
+                    None,
+                    [
+                        get_text(xml_tree, partner_section_xpath + "//Nome"),
+                        get_text(xml_tree, partner_section_xpath + "//Cognome"),
+                    ],
+                )
+            )
```

Comment:
Possiamo usare invece `partner_firstname` come in `16.0`?

Reply:
Era già praticamente fatto in un altro punto, quindi l'ho aggiunto in https://github.com/Borruso/l10n-italy/pull/33